test/src/graph/nodes/Ros2PublishPointsNodeTest.cpp

```diff
@@ -82,4 +83,65 @@ TEST_F(Ros2PublishPointsNodeTest, should_throw_invalid_pipeline_when_no_formatte
 	EXPECT_RGL_SUCCESS(rgl_graph_node_add_child(points, ros2pubWithQos));
 
 	EXPECT_RGL_INVALID_PIPELINE(rgl_graph_run(ros2pubWithQos), "requires a formatted point cloud");
-}
\ No newline at end of file
+}
+
+TEST_F(Ros2PublishPointsNodeTest, should_receive_sent_data)
+{
+	const auto POINT_COUNT = 5;
+	const auto TOPIC_NAME = "rgl_test_pointcloud2";
+	const auto FRAME_ID = "rgl_test_frame_id";
+	const auto NODE_NAME = "rgl_test_node";
+	const auto WAIT_TIME_SECS = 1;
+	const auto MESSAGE_REPEATS = 3;
+	const std::vector<rgl_field_t> fields{DISTANCE_F32};
+	assert(fields.size() == 1 && fields[0] == DISTANCE_F32); // Test is hardcoded to work with distance field only.
+	TestPointCloud input(fields, POINT_COUNT);
+
+	// Create nodes
+	rgl_node_t inputNode = input.createUsePointsNode(), pointcloud2Node = nullptr, format = nullptr;
+	EXPECT_RGL_SUCCESS(rgl_node_points_format(&format, fields.data(), fields.size()));
```

Comment:
I guess going further without success here may not be worth it.
```suggestion
	ASSERT_RGL_SUCCESS(rgl_node_points_format(&format, fields.data(), fields.size()));
```